### PR TITLE
Add project overview page

### PR DESCRIPTION
**Pull request by lentzi90**
This page gives a high level overview of the project, including some graphs and requirements.

It is overlapping a bit with the introduction page, but I hope it will feel more hands-on and explain a bit better how it works, rather than what the project is about.

I also added make targets for linting.

### CODE REVIEW

**Review by kashifest — COMMENTED** [review]
/approve

**Review by tuminoid — COMMENTED** [review]
Lovely addition!

/lgtm
/hold
Giving some time for others to check it.

File: docs/user-guide/src/project-overview.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+![Metal3-CAPI objects](./images/metal3-capi-objects.png)
+
+## How does it work?
+
+Metal3 relies on Ironic for interacting with the physical machines. Ironic in
+turn communicates with [Baseboard Management
+Controllers](https://en.wikipedia.org/wiki/Intelligent_Platform_Management_Interface#Baseboard_management_controller)
+(BMCs) to manage the machines. Ironic can communicate with the BMCs using
+protocols such as Redfish, IPMI, or iDRAC. In this way, it can power on or off
+the machines, change the boot device, and so on.
```
Comment (dtantsur) [inline docs/user-guide/src/project-overview.md:32]: nit: link to the "Ironic in Metal3" page for details

File: docs/user-guide/src/project-overview.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+For more advanced operations, like writing an image to the disk, the [Ironic
+Python Agent](https://opendev.org/openstack/ironic-python-agent) (IPA) is first
```
Comment (dtantsur) [inline docs/user-guide/src/project-overview.md:35]: nit: link to /ironic/ironic-python-agent instead (the git repo is not too helpful)

**Review by dtantsur — DISMISSED** [review]
This is a great addition! A couple of ideas for a potential follow-up inline.

**Comment by lentzi90** [comment]
Thanks for the suggestions!
I have added the links. Also started on a diagram for BareMetalHosts, FirmwareSchema, etc, but it grew out of proportions. I'll leave it for now and hopefully come back to it at a later point.
Let's merge this though :slightly_smiling_face:

**Review by tuminoid — COMMENTED** [review]
/lgtm

File: docs/user-guide/src/project-overview.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+Metal3DataTemplates and related objects. Users can provide metadata and network
+data through these objects. For network data specifically, it is worth
+mentioning the Metal3 [IP address manager (IPAM)](./ipam/introduction.md) that
+can be used to assign IP addresses to the hosts.
+
+## Requirements
+
+- Server(s) with baseboard management capabilities (i.e. Redfish, iDRAC, IPMI,
+  etc.). For development you can use virtual machines with Sushy-tools. More
+  information [here](./bmo/supported_hardware.md).
```
Comment (dtantsur) [inline docs/user-guide/src/project-overview.md:63]: For the future: we should avoid links called "here". I've learned that these are not friendly to accessibility tools (e.g. if a blind user wants the tool to read aloud all links).

**Pull request merged**